src/ecs/SparseArray.hpp

```diff
@@ -106,7 +106,11 @@ namespace ecs
             return _data[pos];
         }
 
-        void erase(sizeType pos) { _data[pos] = std::nullopt; }
+        void erase(sizeType pos)
+        {
+            if (pos < _data.size() && _data[pos])
```

Comment:
is it safe to check the contents of `_data[pos]` on the same condition where you are checking out of bounds ?
Also, why are you checking its content ?